pkg/koyeb/idmapper/app.go

```diff
@@ -0,0 +1,68 @@
+package idmapper
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/koyeb/koyeb-api-client-go/api/v1/koyeb"
+)
+
+// AppMapper is a resolver that translate an app id to a name, and vice-versa.
+type AppMapper struct {
+	ctx    context.Context
+	client *koyeb.APIClient
+	cache  *Cache
+}
+
+// NewAppMapper creates a new instance.
+func NewAppMapper(ctx context.Context, client *koyeb.APIClient) *AppMapper {
+	return &AppMapper{
+		ctx:    ctx,
+		client: client,
+		cache:  NewCache(),
+	}
+}
+
+// GetID translates a name to an id.
+func (mapper *AppMapper) GetID(name string) (string, error) {
+	if IsUUIDv4(name) {
```

Comment:
Should this return an error? I understand we could get there in two cases:
- The name of the app turns out to be a uuid. If uuids are valid names (are they?), I believe it would be a valid case. If uuids are not valid names, we should break here, instead of deceiving the user; or at least log something?
- The caller calls `GetID()` with an app id instead of a name, by mistake. If uuids are valid names, a cache miss is appropriate imo. If uuids ar enot valid names, it is a wrong usage of the function and we could return an error

wdyt?

Reply:
It should stay like this so the user could use `koyeb list instances --app foo` or `koyeb list instances --app 7ab5b7bd-87b2-4045-8261-1d4c9760299a` seamlessly.

Reply:
oooh okay makes sense!

I'm nitpicking there, then, so feel free to implement this idea or not, but should we move that `IsUUIDv4()` in the caller function? Something like:

```go
var filter string = "foo" // could be "7ab5b7bd-87b2-4045-8261-1d4c9760299a" too
id := ""
if IsUUIDv4(filter) {
    id = filter
} else {
    id := GetID(filter)
}
// do something with the id
```

Because `GetID()` takes an app name as parameter, so I expected the distinction "is the argument the user gave me an id or a name?" to be already done

Reply:
@nicoche Updated.

Reply:
Nice!